slf4j/src/main/java/com/google/common/flogger/backend/slf4j/Slf4jLoggerBackend.java

```diff
@@ -0,0 +1,201 @@
+    SimpleMessageFormatter.SimpleLogHandler {
+
+  private final Logger logger;
+
+  private static final MappedLevel OFF_LEVEL = new MappedLevel(new MappedLevel.Predicate() {
+
+    @Override
+    public boolean test(Logger logger) {
+      return false;
+    }
+  }, new MappedLevel.LogAdapter() {
+    @Override
+    public void log(Logger logger, String message, Throwable thrown) {
+      // no-op
+    }
+  });
+
+  private static final MappedLevel TRACE_LEVEL = new MappedLevel(new MappedLevel.Predicate() {
+
+    @Override
+    public boolean test(Logger logger) {
+      return logger.isTraceEnabled();
+    }
+  }, new MappedLevel.LogAdapter() {
+    @Override
+    public void log(Logger logger, String message, Throwable thrown) {
+      logger.trace(message, thrown);
+    }
+  });
+
+  private static final MappedLevel DEBUG_LEVEL = new MappedLevel(new MappedLevel.Predicate() {
+
+    @Override
+    public boolean test(Logger logger) {
+      return logger.isDebugEnabled();
+    }
+  }, new MappedLevel.LogAdapter() {
+    @Override
+    public void log(Logger logger, String message, Throwable thrown) {
+      logger.debug(message, thrown);
+    }
+  });
+  private static final MappedLevel INFO_LEVEL = new MappedLevel(new MappedLevel.Predicate() {
+
+    @Override
+    public boolean test(Logger logger) {
+      return logger.isInfoEnabled();
+    }
+  }, new MappedLevel.LogAdapter() {
+    @Override
+    public void log(Logger logger, String message, Throwable thrown) {
+      logger.info(message, thrown);
+    }
+  });
+  private static final MappedLevel WARN_LEVEL = new MappedLevel(new MappedLevel.Predicate() {
+
+    @Override
+    public boolean test(Logger logger) {
+      return logger.isWarnEnabled();
+    }
+  }, new MappedLevel.LogAdapter() {
+    @Override
+    public void log(Logger logger, String message, Throwable thrown) {
+      logger.warn(message, thrown);
+    }
+  });
+  private static final MappedLevel ERROR_LEVEL = new MappedLevel(new MappedLevel.Predicate() {
+
+    @Override
+    public boolean test(Logger logger) {
+      return logger.isErrorEnabled();
+    }
+  }, new MappedLevel.LogAdapter() {
+    @Override
+    public void log(Logger logger, String message, Throwable thrown) {
+      logger.error(message, thrown);
+    }
+  });
+
+  Slf4jLoggerBackend(Logger logger) {
+    if (logger == null) {
+      throw new NullPointerException("logger is required");
+    }
+    this.logger = logger;
+  }
+
+  private static MappedLevel mapLevel(Level level) {
+    int requestedLevel = level.intValue();
+    if( requestedLevel < Level.FINE.intValue() ) {
+      return TRACE_LEVEL;
+    }
+
+    if( requestedLevel < Level.INFO.intValue() ) {
+      return DEBUG_LEVEL;
+    }
+
+    if( requestedLevel < Level.WARNING.intValue() ) {
+      return INFO_LEVEL;
+    }
+
+    if( requestedLevel < Level.SEVERE.intValue() ) {
+      return WARN_LEVEL;
+    }
+
+    if( requestedLevel < Level.OFF.intValue() ) {
```

Comment:
This will always be true for Flogger, since it's not permitted to use a level with a value equal to "OFF" when calling a log statement. OFF and ALL are sentinel values for configuration purposes, not allowed levels for logging. The check I think you'd want to make here is "is slf4J configured to disable all logging" (if such a check is even possible in slf4j) and should not read the given log level for this final check.

Reply:
Was wondering about that, thanks for clarifying.  Latest commit adds documentation around the level mapping process and restricts Level.OFF and Level.ALL from being used in logging.